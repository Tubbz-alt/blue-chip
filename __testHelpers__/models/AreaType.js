import BaseModel from "../../src/BaseModel";
import Spec from "./Spec";

export default class AreaType extends BaseModel {}
