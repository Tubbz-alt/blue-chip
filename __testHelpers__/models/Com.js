import BaseModel from "../../src/BaseModel";
import SpecDetail from "./SpecDetail";

export default class Com extends BaseModel {
  static get belongsTo() {
    return [SpecDetail];
  }
}
