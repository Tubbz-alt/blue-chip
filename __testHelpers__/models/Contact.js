import BaseModel from "../../src/BaseModel";
import VendorContact from "./VendorContact";

export default class Contact extends BaseModel {
  static get belongsTo() {
    return [VendorContact];
  }
}
