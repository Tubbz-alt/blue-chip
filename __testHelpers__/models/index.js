import Checklist from "./Checklist";
import Task from "./Task";
import Contact from "./Contact";
import PurchaseOrder from "./PurchaseOrder";
import PurchaseOrderContact from "./PurchaseOrderContact";
import Vendor from "./Vendor";
import VendorContact from "./VendorContact";
import Spec from "./Spec";
import SpecDetail from "./SpecDetail";
import SpecDetailCom from "./SpecDetailCom";

export {
  Checklist,
  Task,
  Contact,
  PurchaseOrder,
  PurchaseOrderContact,
  Vendor,
  VendorContact,
  Spec,
  SpecDetail,
  SpecDetailCom
};
