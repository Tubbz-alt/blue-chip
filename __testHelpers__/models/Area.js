import BaseModel from "../../src/BaseModel";
import Spec from "./Spec";
import AreaType from "./AreaType";
import AreaRoom from "./AreaRoom";

export default class Area extends BaseModel {
  static get belongsTo() {
    return [Spec, AreaType];
  }
  static get hasMany(){
    return [AreaRoom];
  }
}
