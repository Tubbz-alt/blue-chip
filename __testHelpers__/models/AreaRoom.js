import BaseModel from "../../src/BaseModel";
import Area from "./Area";

export default class AreaRoom extends BaseModel {
  static get belongsTo(){
    return [Area]
  }
}
