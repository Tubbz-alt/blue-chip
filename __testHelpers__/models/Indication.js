import BaseModel from "../../src/BaseModel";
import Patient from "./Patient";

export default class Indication extends BaseModel {
  static get hasMany() {
    return [Patient];
  }
}
