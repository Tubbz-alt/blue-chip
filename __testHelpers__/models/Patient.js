import BaseModel from "../../src/BaseModel";
import SettingOfUse from "./SettingOfUse";
import Indication from "./Indication";

export default class Patient extends BaseModel {
  static get belongsTo() {
    return [SettingOfUse, Indication];
  }
}
