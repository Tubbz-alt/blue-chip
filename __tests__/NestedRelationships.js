import checklistResources from "../__testHelpers__/fixtrues/checklistsAndTasksNormalized";
import pOResources from "../__testHelpers__/fixtrues/purchaseOrdersAndOthersNormalized";
import specResources from "../__testHelpers__/fixtrues/specsNormalized";
import {Checklist, PurchaseOrderContact, Spec} from "../__testHelpers__/models";

describe("Nested resourse", () => {
  test("includes with hasMany.belongsTo", () => {
    const checklists = Checklist.query(checklistResources)
      .all()
      .includes(["tasks.user"])
      .toObjects();
    expect(checklists).toMatchSnapshot();
  });

  test("includes with belongsTo.belongsTo", () => {
    const purchaseOrderContacts = PurchaseOrderContact.query(pOResources)
      .includes(["vendorContact.contact"])
      .toObjects();
    expect(purchaseOrderContacts).toMatchSnapshot();
  });

  test("includes with hasMany.hasMany", () => {
    const specs = Spec.query(specResources)
      .where({id: [11]})
      .includes(["specDetails.coms"])
      .toObjects();
    expect(specs).toMatchSnapshot();
  });

  test("belongsTo, hasMany.[hasMany, hasMany, belonngsTo]", () => {
    const specs = Spec.query(specResources)
      .where({id: [11]})
      .includes(["specCategory", "specDetails.[coms, roomTypes, user]"])
      .toObjects();
    expect(specs).toMatchSnapshot();
  });
});
