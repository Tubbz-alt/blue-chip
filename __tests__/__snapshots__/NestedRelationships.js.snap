// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nested resourse returns a snapshot including users 1`] = `
Array [
  Object {
    "id": 3,
    "name": "QA Checklist Rest",
    "tasks": Array [
      Object {
        "description": "QA Checklist Rest 5",
        "id": 5,
      },
      Object {
        "description": "QA Checklist Rest 6",
        "id": 6,
      },
    ],
  },
  Object {
    "id": 1,
    "name": "Onboarding Rest",
    "tasks": Array [
      Object {
        "description": "Onboarding REST 1",
        "id": 1,
        "user": Object {
          "id": 1,
          "name": "Joe",
        },
      },
      Object {
        "description": "Onboarding REST 2",
        "id": 2,
      },
    ],
  },
  Object {
    "id": 2,
    "name": "Project Audit Rest",
    "tasks": Array [
      Object {
        "description": "Project Audit Rest 3",
        "id": 3,
      },
      Object {
        "description": "Project Audit Rest 4",
        "id": 4,
      },
    ],
  },
]
`;
